citesphere/src/main/java/edu/asu/diging/citesphere/core/util/model/impl/CitationHelper.java

```diff
@@ -122,6 +123,18 @@ public void updateCitation(ICitation citation, CitationForm form, IUser user) {
             }
         }
     }
+    
+    @Override
+    public void updateCitation(ICitation citation, String collection, IUser iUser) {
+        List<String> collections = citation.getCollections();
+        if (collections != null) {
+            collections.add(collection);
+        } else {
+            collections = new ArrayList<String>();
+            collections.add(collection);
+        }
```

Comment:
instead test for null, and if collections is null, `setCollections(new ArrayList)`, then just add collection to the list. No else is needed and no extra call to `setCollections`.